src/full_node/blockchain.py

```diff
@@ -411,7 +411,7 @@ async def validate_block_body(self, block: Union[FullBlock, UnfinishedBlock]) ->
             return None  # This means the sub-block is valid
 
         # 2. For blocks, foliage block, transaction filter, transactions info must not be empty
-        if block.foliage_block is None or block.transactions_filter is None or block.transactions_info is None:
```

Comment:
I believe filter should always be there as well

Reply:
Oh, the filter is computed from the transactions generator. once again you're right